### ISSUE TITLE
🐛 BUG: Astro parser error with Base Component

### ISSUE THREAD

**Issue by drwpow** [snowpackjs/astro#1133]
### What package manager are you using?

yarn

### What operating system are you using?

Mac

### Describe the Bug

This works (when there’s no HTML):

```
---
import Base from '../layouts/BaseLayout.astro';
---
<Base>
  Works
</Base>
```

This doesn’t when HTML is added:

```
---
import Base from '../layouts/BaseLayout.astro';
---
<Base>
  <div>Doesn't work</div>
</Base>
```

⚠️ **However!**

This also works:

```
---
import BaseLayout from '../layouts/BaseLayout.astro';
---
<BaseLayout>
  <div>Doesn't work</div>
</BaseLayout>
```

Looking more closely, the codegen output is wildly different based on all these minor settings. It seems that there is some obscure parser error here.

### Steps to Reproduce

https://codesandbox.io/s/fragrant-sound-tnh52?file=/src/pages/index.astro

### Link to Minimal Reproducible Example (Optional)

https://codesandbox.io/s/fragrant-sound-tnh52?file=/src/pages/index.astro

**Comment by jasikpark** [comment]
I've been experiencing something similar while testing out the beta prettier plugin - there's potentially just an error in the plugin, but when I format
```
<Footer></Footer>
```
i get out
```
<footer></footer>
```
which seems like the parser might be passing components into the html parser, which would find mixed-case html elements perfectly valid!

**Comment by natemoo-re** [comment]
The team has decided to close out issues that have been confirmed as fixed by [astro@0.21.0-next.0](https://github.com/snowpackjs/astro/releases/tag/astro%400.21.0-next.0), [astro@0.21.0-next.1](https://github.com/snowpackjs/astro/releases/tag/astro%400.21.0-next.1), or [astro@0.21.0-next.2](https://github.com/snowpackjs/astro/releases/tag/astro%400.21.0-next.2). Our hope is that this will help the [v0.21 milestone](https://github.com/snowpackjs/astro/milestone/1) remain as actionable as possible.

To verify that this issue has been fixed, you may
- Open a new [`astro@next` project on StackBlitz](https://astro.new/minimal@next)
- Locally upgrade to the `astro@next` preview
    ```shell
    npm i astro@next
    # If using Framework renderers...
    npm i @astrojs/renderer-name@next
    ```

If you think we made a mistake, **please reopen this issue**!

If you run into any other problems with `astro@next`, please [open a new issue](https://github.com/snowpackjs/astro/issues/new?assignees=&labels=0.21&template=---01-bug-report.yml&title=%F0%9F%90%9B+BUG%3Aastro@next).
